### PR TITLE
Cleanup tests

### PR DESCRIPTION
**Pull request by res0nance**
- Cleanup some tests with using the older now NOOPed `doYank()` method
- Rewrite the delay calculation in RateLimitBranchPropertyTest to not be truncated.
- Fix the delay to not be an overly bloated number

Running it in CI to try to reproduce all the test flakes

### CODE REVIEW

File: src/test/java/jenkins/branch/NoTriggerBranchPropertyTest.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void smokes() throws Exception {
         sampleRepo.git("add", "server");
         sampleRepo.git("commit", "--all", "--message=release");
         MultiBranchImpl stuff = r.jenkins.createProject(MultiBranchImpl.class, "stuff");
-        BranchSource branchSource = new BranchSource(new GitSCMSource(null, sampleRepo.toString(), "", "*", "", false));
+        BranchSource branchSource = new BranchSource(new GitSCMSource(sampleRepo.toString()));
```
Comment (res0nance) [inline src/test/java/jenkins/branch/NoTriggerBranchPropertyTest.java]: This is somehow not equivalent hence why the test breaks.

File: src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java [hunk 2]
```diff
@@ -286,11 +287,12 @@ public void rateLimitsConcurrentBuilds() throws Exception {
             FreeStyleBuild secondBuild = startCondition.get();
             // it can take more than the requested delay... that's ok, but it should not be
             // more than 500ms longer (i.e. 5 of our Queue.maintain loops above)
+            final long delay = (long)(60.f * 60.f / rate * 1000);
             assertThat("At least the rate implied delay but no more than 500ms longer",
                     secondBuild.getStartTimeInMillis() - firstBuild.getStartTimeInMillis(),
                     allOf(
-                            greaterThanOrEqualTo(60 * 60 / rate * 1000L - 200L),
-                            lessThanOrEqualTo(60 * 60 / rate * 1000L * 500L)
```
Comment (res0nance) [inline src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java]: The `*` in the less than condition from the comment seems to be a typo

**Comment by res0nance** [comment]
I tested this locally and i got a mind boggling result
```
[ERROR] Failures:
[ERROR]   RateLimitBranchPropertyTest.rateLimitsBlockBuilds_maxRate:133->rateLimitsBlockBuilds:201 At least the rate implied delay but no more than 500ms longer
Expected: (a value equal to or greater than <3400L> and a value less than or equal to <4100L>)
     but: a value equal to or greater than <3400L> <1718L> was less than <3400L>
[ERROR]   RateLimitBranchPropertyTest.rateLimitsBlockBuilds_medRate:138->rateLimitsBlockBuilds:201 At least the rate implied delay but no more than 500ms longer
Expected: (a value equal to or greater than <7000L> and a value less than or equal to <7700L>)
     but: a value equal to or greater than <7000L> <5332L> was less than <7000L>
[INFO]
[ERROR] Tests run: 10, Failures: 2, Errors: 0, Skipped: 0
```

File: src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java [hunk 3]
```diff
@@ -169,13 +170,14 @@ public void rateLimitsBlockBuilds(int rate) throws Exception {
             );
             assertThat(master.isInQueue(), is(false));
             assertThat(master.getQueueItem(), nullValue());
+            assertThat(master.getBuilds().getLastBuild(), isNotNull());
+            long startTime = master.getBuilds().getLastBuild().getTimeInMillis();
```
Comment (res0nance) [inline src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java]: This is probably why it flakes all the time lastBuild is not null so it uses the time of the last build to throttle
https://github.com/jenkinsci/branch-api-plugin/blob/a1c604b25e6d7aa704f82a0138699cee32388d07/src/main/java/jenkins/branch/RateLimitBranchProperty.java#L507-L530

**Comment by bitwiseman** [comment]
I watch with interest...

**Comment by res0nance** [comment]
> I watch with interest...

Now i always run into a different error. 😅

**Comment by res0nance** [comment]
Huzzah it worked

**Comment by res0nance** [comment]
It shouldn't be flaky as it stopped flaking on my local machine the one with
```
[ERROR] Failures:
[ERROR]   RateLimitBranchPropertyTest.rateLimitsBlockBuilds_maxRate:133->rateLimitsBlockBuilds:201 At least the rate implied delay but no more than 500ms longer
Expected: (a value equal to or greater than <3400L> and a value less than or equal to <4100L>)
     but: a value equal to or greater than <3400L> <1718L> was less than <3400L>
[ERROR]   RateLimitBranchPropertyTest.rateLimitsBlockBuilds_medRate:138->rateLimitsBlockBuilds:201 At least the rate implied delay but no more than 500ms longer
Expected: (a value equal to or greater than <7000L> and a value less than or equal to <7700L>)
     but: a value equal to or greater than <7000L> <5332L> was less than <7000L>
[INFO]
[ERROR] Tests run: 10, Failures: 2, Errors: 0, Skipped: 0
```
So at least for this flake it should stop, it was root caused in my previous comment as well. So I expect the flaking for this particular test to stop

**Pull request merged**